### PR TITLE
Playwright: set up magic link spec.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR moves the Magic Link tests to a standalone spec.

Key changes:
- schedule Magic Links spec to run with each Pre-Release test run.
- magic links spec to run with default test user.

#### Testing instructions

- [x] mobile
- [x] desktop
- [x] pre-release

Related to #

### CODE REVIEW

File: test/e2e/specs/specs-playwright/wp-auth__magic-link.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { Page } from 'playwright';
+
+describe( DataHelper.createSuiteTitle( 'Authentication: Magic Link' ), function () {
+	const inboxId = DataHelper.config.get( 'defaultUserInboxId' ) as string;
+	// This method of obtaining the email address of the test user is temporary.
+	// The encrypted configuration file needs a revamp and have the format standardized
+	// and when that is copmlete, this can be replaced with an appropriate query method
+	// to extract the email address of the test user.
+	// See https://github.com/Automattic/wp-calypso/issues/55694.
+	const email = `main.${ inboxId }@mailosaur.io`;
```
Comment (worldomonation) [inline test/e2e/specs/specs-playwright/wp-auth__magic-link.ts:15]: I've written it in the code, but I'd like to expand on this issue.

Currently, our config file is bit of a mess, as noted in the linked issue https://github.com/Automattic/wp-calypso/issues/55694.

We have two interspersed formats that specify test accounts:

```json
	"testAccounts": {
    },
    "accounts": {
    }

```

Each with slightly different schemas:

```json
"testAccounts": {
   "defaultUser": [ username, password, optional test site ],
}
```

```json
"accounts": {
  {
    username: <username>,
    password: <password>,
    site: <site>,
    features: [ feature_1 ],
}
```

All of e2e tests currently use the `testAccounts` schema but the helper method in `DataHelper` doesn't have a method to obtain the Mailosaur Inbox ID of a particular user - only method present is to obtain the credentails and the primary test site address.

The linked task is to refactor/standardize the schema, after which methods like `DataHelper.getInboxForUser` or `DataHelper.getPrimarySiteURL` can be implemented.

File: test/e2e/specs/specs-playwright/wp-auth__magic-link.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+			subject: 'Log in to WordPress.com',
+		} );
+		const links = await emailClient.getLinksFromMessage( message );
+		magicLink = links.find( ( link: string ) => link.includes( 'wpcom_email_click' ) ) as string;
+		expect( magicLink ).toBeDefined();
+	} );
+
+	it( 'Log in using magic link', async function () {
+		const loginPage = new LoginPage( page );
+		await Promise.all( [ page.waitForNavigation(), loginPage.followMagicLink( magicLink ) ] );
```
Comment (dpasque) [inline test/e2e/specs/specs-playwright/wp-auth__magic-link.ts:47]: Small nit, but I don't think you need this waitForNavigation as there's one in `followMagicLink`

Unless you want to validate that we specific navigate to the home page for the user? That might not be a bad validation to add, but it's not entirely necessary.

**Review by dpasque — APPROVED** [review]
One optional comment, otherwise LGTM!

+1 on cleaning up config soon

File: test/e2e/specs/specs-playwright/wp-auth__magic-link.ts (hunk 2 above)
Comment (worldomonation) [reply test/e2e/specs/specs-playwright/wp-auth__magic-link.ts:47]: You're right - i was not paying attention and thought this was the Gutenboarding Language PR, and have accidentally merged it 🤦 

I'll remove the unnecessary `waitForNavigation` in a tiny followup PR.

Thanks for always reviewing the code very carefully. I appreciate it.

Comment (worldomonation) [reply test/e2e/specs/specs-playwright/wp-auth__magic-link.ts:47]: PR for the nit: https://github.com/Automattic/wp-calypso/pull/56800

**Pull request merged**